### ISSUE TITLE
Use Vocabulary Terms

### ISSUE THREAD

**Issue by dselman** [accordproject/template-engine#10]
The Template Engine should use terms from the vocabulary when generating text. This requires associating a vocabulary with a template.

**Comment by Abhii67534** [comment]
Is this issue solved?

**Comment by sanketshevkar** [comment]
I don't think so. @dselman can you please confirm?

**Comment by dselman** [comment]
No, this is not yet resolved. It really only manifests itself for enums.

**Comment by Shubh-Raj** [comment]
Hi @dselman ,

I've investigated this issue and confirmed it **still exists**. The Template Engine currently does not use vocabulary terms when generating text for enums.

### Current Behavior
When generating output from templates like `insurance_fr`, enum values are rendered as raw identifiers:
- `"CAR, ACCESSORIES et SPARE_PARTS"` instead of localized terms
- `"REPAIRING"` / `"REPLACING"` instead of `"réparation"` / `"remplacement"`

### Root Cause
Looking at [TemplateMarkInterpreter.ts](cci:7://file:///home/shubhraj/OpenSource/template-engine/src/TemplateMarkInterpreter.ts:0:0-0:0), the enum drafting logic treats enums as plain strings without vocabulary lookup:

```typescript
// we want to draft Enums as strings, not objects
const drafter = getDrafter(type && type.isEnum() ? 'String' : context.elementType);
```
The Concerto ecosystem has @accordproject/concerto-vocabulary for translating model terms, but this capability is not integrated into the template engine.

### Proposed Solution

1. Accept optional VocabularyManager in TemplateMarkInterpreter constructor
2. Add locale to GenerationOptions to specify the target language
3. Look up vocabulary terms when drafting enum values using VocabularyManager.resolveTerm()
4. Graceful fallback to raw enum values if no vocabulary is provided

I am interested in working on this.
